examples/foundational/19-openai-realtime-beta.py

```diff
@@ -126,7 +124,24 @@ async def main():
         # llm.register_function(None, fetch_weather_from_api)
         llm.register_function("get_current_weather", fetch_weather_from_api)
 
-        context = OpenAILLMContext([{"role": "user", "content": "Say hello!"}], tools)
+        # Create a standard OpenAI LLM context object using the normal messages format. The
+        # OpenAIRealtimeBetaLLMService will convert this internally to messages that the
+        # openai WebSocket API can understand.
+        context = OpenAILLMContext(
+            [{"role": "user", "content": "Say hello!"}],
+            # [{"role": "user", "content": [{"type": "text", "text": "Say hello!"}]}],
```

Comment:
Should `type` be `input_text`? Same below

Reply:
Should we also say "OpenAI Realtime message format is also supported"? Which I think is the commented code.

Reply:
No, we don't support a "realtime message format" because the current version of the realtime API doesn't let you add messages to the context in a flexible way.

We're not sure how much of this is intentional, and how much the API is going to change during the beta. Developers who really want to directly add messages via the realtime API can call `llm.send_client_event()`. But adding more than one message at a time via events has unpredictable results.

Right now we're trying to support just the traditional OpenAI messages format. And we only support setting up the context at the beginning of the session. If the messages list or context changes, we disconnect from the websocket, reconnect, and reload the messages by packing them together into one single large message.

The code that's commented out is to test two different ways of specifying "content" in the traditional OpenAI messages format.

Reply:
Ah, perfect! Thanks, it looked very similar to the new format that's why I was confused!